memgpt/server/rest_api/routers/v1/agents.py

```diff
@@ -0,0 +1,238 @@
+from typing import TYPE_CHECKING
+from fastapi import APIRouter, Depends, HTTPException, JSONResponse, status
+
+from memgpt.server.rest_api.utils import get_current_user, get_current_interface, get_memgpt_server
+from memgpt.models.pydantic_models import  AgentStateModel, EmbeddingConfigModel, LLMConfigModel
+from memgpt.server.schemas.agents import AgentCommandResponse, GetAgentResponse, AgentRenameRequest, CreateAgentRequest, CreateAgentResponse, ListAgentsResponse
+
+if TYPE_CHECKING:
+    from uuid import UUID
+    from memgpt.orm.user import User
+    from memgpt.server.schemas.agents import AgentCommandRequest
+    from memgpt.server.rest_api.interface import QueuingInterface
+    from memgpt.server.server import SyncServer
+
+router = APIRouter(prefix="/agents", tags=["agents"])
+
+@router.get("/", response_model=ListAgentsResponse)
+def list_agents(
+    actor: "User" = Depends(get_current_user),
+    interface: "QueuingInterface" = Depends(get_current_interface),
+    server: "SyncServer" = Depends(get_memgpt_server),
+):
+    """
+    List all agents associated with a given user.
+
+    This endpoint retrieves a list of all agents and their configurations associated with the specified user ID.
+    """
+    interface.clear()
+    return server.list_agents(user_id=actor._id)
+
+@router.post(response_model=CreateAgentResponse)
+def create_agent(
+    agent: CreateAgentRequest,
+    actor: "User" = Depends(get_current_user),
+    interface: "QueuingInterface" = Depends(get_current_interface),
+    server: "SyncServer" = Depends(get_memgpt_server),
+):
+    """
+    Create a new agent with the specified configuration.
+    """
+    interface.clear()
+
+    # Parse request
+    # TODO: don't just use JSON in the future
+    human_name = request.config["human_name"] if "human_name" in request.config else None
```

Comment:
@sarahwooders I'm bringing this over but all of this should be part of the schema - I can merge your branch into this one and use `AgentCreate` (I think that's what you said it'll be called) instead of this, or we can just reconcile during the merge into v1